script.js

```diff
@@ -12,3 +12,253 @@ mobileLinks.addEventListener('click', () => {
+
+    a.textContent = tag;
+
+    li.appendChild(a);
+    ul.appendChild(li);
+  });
+
+  return ul;
+}
+function createH4(h) {
+  const h4 = document.createElement('h4');
+  h4.textContent = h;
+  return h4;
+}
+
+function createp(para) {
+  const p = document.createElement('p');
+  p.textContent = para;
+  return p;
+}
+
+function createButton(cbutton) {
+  const button = document.createElement('button');
+  button.classList.add('button-orange', 'is-link');
+  button.textContent = cbutton;
+  return button;
+}
+function createArticle(articleContent) {
+  const article = document.createElement('article');
+  article.classList.add('works');
+
+  const worksDescription = document.createElement('div');
+  worksDescription.classList.add('works-description');
+
+  const h4tag = createH4(articleContent.title);
+  const para = createp(articleContent.paragraph);
+  const links = createArticleLinks(articleContent.technologyArticle);
+  const buttonOrange = createButton(articleContent.prjBtn);
+
+  worksDescription.appendChild(h4tag);
+  worksDescription.appendChild(para);
+  worksDescription.appendChild(links);
+  worksDescription.appendChild(buttonOrange);
+
+  article.appendChild(worksDescription);
+
+  return article;
+}
+
+function createArticles(content) {
+  const grid = document.querySelector('.works-grid');
+
+  content.forEach((article) => {
+    const art = createArticle(article);
+    grid.appendChild(art);
+  });
+}
+
+createArticles(articles);
+
+// modal popup
+const modal = document.querySelector('.dsk-width');
+const modalBtns = document.querySelectorAll('.button-orange');
+const closeBtn = document.getElementsByClassName('close-btn')[0];
+
+modalBtns.forEach((modalBtn) => {
+  modalBtn.addEventListener('click', () => {
+    modal.style.display = 'block';
+  });
+});
+closeBtn.addEventListener('click', () => {
+  modal.style.display = 'none';
+});
+
+const details = [
+  {
+    title: 'Keeping track of hundreds of components website',
+    featuredImage: {
+      src: 'assets/Snapshoot Portfolio.png',
+      alt: 'featured image',
+    },
+    technologies: ['html', 'bootstrap', 'Ruby on rails'],
+    description: `Lorem Ipsum is simply dummy text of the printing and typesetting
+              industry. Lorem Ipsum has been the industry's standard dummy text
+              ever since the 1500s, when an unknown printer took a galley of
+              type and scrambled it 1960s with the releaLorem Ipsum is simply
+              dummy text of the printing and typesetting industry. Lorem Ipsum
+              has been the industry's standard dummy text ever since the 1500s,
+              when an unknown printer took a galley of type and scrambled it
+              1960s with the releax map lapora verita.`,
+    liveVersion: 'See live',
+    source: 'See Source',
+  },
+];
+
+const desktopTitle = document.querySelector('.dsk-popup-title');
+const desktopP = document.querySelector('.dsk-txt');
+
+details.forEach((detail) => {
```

Comment:
- [ ] Please kindly ensure that the modal popup is populated with data from the projects array created previously.